src/jit/importer.cpp

```diff
@@ -7522,7 +7522,9 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
             }
         }
 
-        if (!bIntrinsicImported)
+        // Don't create cast for a variable on the stack if it is intrinsic or tail call.
+        // We assume that the value will not be physically represented there.
+        if (!bIntrinsicImported && !(tailCall && canTailCall))
```

Comment:
It can't happen if checkForSmallType  == false (checkForSmallType  = opts.IsJit64Compat() || opts.IsReadyToRun();). So you can see it only in R2R or in JIT64Compat mode. 
It is possible to set this flag to true, in this case you will see the assert error in coreclr.
I think it is impossible to reproduce this failure on coreclr without changing source code.